src/thunderJS.js

```diff
@@ -62,6 +59,7 @@ const resolve = (result, args) => {
 
 const thunder = options => ({
   options,
+  api: API(options),
```

Comment:
Adding this here has the effect that `API` is now reinitialised upon _every_ JSON RPC call (and not reused within the same session as before). That doesn't seem desirable (unless there really isn't another way)

Reply:
Not sure how to achieve this. Before the API was indeed reused in same session. But not only that, it was reused between all sessions/instances of thunderjs. Which is also not good since API might need different options for different sessions/instances.